### ISSUE TITLE
[QUESTION] Learning Rate Scaling in DINOv2 Continued Pretraining

### ISSUE THREAD

**Issue by blackpearl006** [lightly-ai/lightly-train#332]
### Question

Is there automatic learning rate scaling applied during DINOv2 continued pretraining? What is the logic behind it, and what is the recommended default learning rate?

### What I've Tried

I experimented with standard learning rates (1e-4 and 1e-5) but observed that the effective learning rate appears to be scaled down automatically. 

Sample LR rate during training for 200 epochs when started with value 0.004

![optim_args documentation](https://github.com/user-attachments/assets/bcd23775-a767-4733-bd96-a8bc32784f7d)

When using higher learning rates like the configuration below, they seem to be scaled down to approximately 1e-5:

```python
optim = 'adamw',
optim_args = {
    "betas": [0.9, 0.999],
    "eps": 1e-08,
    "lr": 0.004,
    "weight_decay": 0.04
}
```

### Context & Documentation

According to the [optim_args documentation](https://docs.lightly.ai/train/stable/python_api/lightly_train.html#:~:text=on%20the%20method.-,optim_args%3A,optim_args%3D%7B%22lr%22%3A%20float%2C%20%22momentum%22%3A%20float%2C%20%22weight_decay%22%3A%20float%7D,-transform_args%3A):

![Additional documentation](https://github.com/user-attachments/assets/1767d940-ebad-4936-ba2f-e04405539157)

### Expected Information

- Is there an automatic learning rate scaling mechanism in place?
- If so, what is the scaling formula or logic?
- What learning rate values should be specified in `optim_args` given this scaling behavior?
- What are the recommended default learning rates for DINOv2 continued pretraining?

**Comment by stegmuel** [comment]
Hello,

Yes, similar to the official implementation the learning rate is scaled w.r.t. a reference batch size (1024).
The rationale behind this is to avoid having to change the learning when the batch size is changed.
This is implemented [here](https://github.com/lightly-ai/lightly-train/blob/5300ddb2a95aaada36f3afa60330cbe5ee7b281c/src/lightly_train/_methods/dinov2/dinov2.py#L554).
Learning rate in the range [4e-3, 1e-4] tend to work well.
I would start with the default value.

**Comment by liopeer** [comment]
> Is there an automatic learning rate scaling mechanism in place?

The scaling method is defined via the arguments I mentioned in your other issue: https://github.com/lightly-ai/lightly-train/issues/329

> If so, what is the scaling formula or logic?

For the scaling we use a `CosineWarmupScheduler` defined here:
https://github.com/lightly-ai/lightly-train/blob/5300ddb2a95aaada36f3afa60330cbe5ee7b281c/src/lightly_train/_methods/dinov2/dinov2.py#L576-L585

and all your configurable arguments are sitting here:

https://github.com/lightly-ai/lightly-train/blob/5300ddb2a95aaada36f3afa60330cbe5ee7b281c/src/lightly_train/_methods/dinov2/dinov2.py#L156-L164

> What learning rate values should be specified in optim_args given this scaling behavior?

> What are the recommended default learning rates for DINOv2 continued pretraining?

I'd always go with the defaults and only adjust if you're unhappy with the results. There are also some additional comments in the code I linked.
